tests/devices/test_default_mixed.py

```diff
@@ -39,105 +73,426 @@ def test_expected_state(self, nr_wires, index):
         rho = np.zeros((2 ** nr_wires, 2 ** nr_wires))
         rho[index, index] = 1
         rho = np.reshape(rho, [2] * (2 * nr_wires))
-        dev = qml.device('default.mixed', wires=nr_wires)
+        dev = qml.device("default.mixed", wires=nr_wires)
 
         assert np.allclose(rho, dev._create_basis_state(index))
 
 
-@pytest.mark.parametrize("nr_wires", [1, 2, 3])
+@pytest.mark.parametrize("nr_wires", [2, 3])
 class TestState:
     """Tests for the method `state()`, which retrieves the state of the system"""
-    # Note: These tests need to be extended once we can output non-basis states
 
     def test_shape(self, nr_wires):
         """Tests that the state has the correct shape"""
-        dev = qml.device('default.mixed', wires=nr_wires)
+        dev = qml.device("default.mixed", wires=nr_wires)
 
         assert (2 ** nr_wires, 2 ** nr_wires) == np.shape(dev.state)
 
     def test_init_state(self, nr_wires):
         """Tests that the state is |0...0><0...0| after initialization of the device"""
         rho = np.zeros((2 ** nr_wires, 2 ** nr_wires))
         rho[0, 0] = 1
-        dev = qml.device('default.mixed', wires=nr_wires)
+        dev = qml.device("default.mixed", wires=nr_wires)
 
         assert np.allclose(rho, dev.state)
 
-
-@pytest.mark.parametrize("nr_wires", [1, 2, 3])
+    @pytest.mark.parametrize("op", all_ops)
+    def test_state_after_op(self, nr_wires, op):
```

Comment:
While parametrization is great for test data, I would generally avoid parametrizations that result in a single test testing different _logic_. E.g., if you find yourself needing to have if statements in a test to work with different parametrized data types, this is normally a sign that each if statement should be a separate test.

In this case, I would have a separate test for the two mode gate, the channels, and the non-parametrized gate.